packages/backend/src/utils/inferenceUtils.ts

```diff
@@ -0,0 +1,147 @@
+    output = providers.find(engine => engine.providerId === providerId);
+  } else {
+    // Have a preference for a podman engine
+    output = providers.find(engine => engine.connection.type === 'podman');
+    if (output === undefined) {
+      output = providers[0];
+    }
+  }
+  if (output === undefined) throw new Error('cannot find any started container provider.');
+  return output;
+}
+
+/**
+ * Given an image name, it will return the ImageInspectInfo corresponding. Will raise an error if not found.
+ * @param connection
+ * @param image
+ * @param callback
+ */
+export async function getImageInfo(
+  connection: ContainerProviderConnection,
+  image: string,
+  callback: (event: PullEvent) => void,
+): Promise<ImageInfo> {
+  console.debug(`get image ${image} with connection ${connection.name}.`);
+
+  let imageInfo: ImageInfo;
+  try {
+    // Pull image
+    await containerEngine.pullImage(connection, image, callback);
+    // Get image inspect
+    imageInfo = (
+      await containerEngine.listImages({
+        provider: connection,
+      } as ListImagesOptions)
+    ).find(imageInfo => imageInfo.RepoTags?.some(tag => tag === image));
+  } catch (err: unknown) {
+    console.warn('Something went wrong while trying to get image inspect', err);
+    throw err;
+  }
+
+  if (imageInfo === undefined) throw new Error(`image ${image} not found.`);
+
+  return imageInfo;
+}
+
+/**
+ * Given an {@link InferenceServerConfig} and an {@link ImageInfo} generate a container creation options object
+ * @param config the config to use
+ * @param imageInfo the image to use
+ */
+export function GenerateContainerCreateOptions(
+  config: InferenceServerConfig,
+  imageInfo: ImageInfo,
+): ContainerCreateOptions {
+  if (config.modelsInfo.length === 0) throw new Error('Need at least one model info to start an inference server.');
+
+  if (config.modelsInfo.length > 1) {
+    throw new Error('Currently the inference server does not support multiple models serving.');
+  }
+
+  const modelInfo = config.modelsInfo[0];
+
+  if (modelInfo.file === undefined) {
+    throw new Error('The model info file provided is undefined');
+  }
+
+  return {
+    Image: imageInfo.Id,
+    Detach: true,
+    ExposedPorts: { [`${config.port}`]: {} },
+    HostConfig: {
+      AutoRemove: false,
+      Mounts: [
+        {
+          Target: '/models',
+          Source: modelInfo.file.path,
+          Type: 'bind',
+        },
+      ],
+      PortBindings: {
+        '8000/tcp': [
+          {
+            HostPort: `${config.port}`,
+          },
+        ],
+      },
+    },
+    HealthCheck: {
+      // must be the port INSIDE the container not the exposed one
+      Test: ['CMD-SHELL', `curl -sSf localhost:8000/docs > /dev/null`],
+      Interval: 1_000_000_000 * 15, // 15s
+      Retries: 4 * 5, // 20 * 15s = 300s = 5 minutes
+    },
+    Labels: {
+      ...config.labels,
+      [LABEL_INFERENCE_SERVER]: JSON.stringify(config.modelsInfo.map(model => model.id)),
```

Comment:
The server is started only for the first model, why labels are about all models

Reply:
For now we only support one model. But in the future the Inference server can support several models see https://github.com/redhat-et/locallm/pull/72. So using something like `MODEL_ID` does not make sense as we want to be able to link more than one model to a single Inference server;

Reply:
So if you find such a container you will think it supports several models where it has been started with a single one

Reply:
https://github.com/projectatomic/ai-studio/blob/fa1fd271259eb9a94c9c357cc2b833afbfa43214/packages/backend/src/utils/inferenceUtils.ts#L101-L103

If you go just a little higher at line 103, you can see why I am only using the first at line 105, because currently we do not support more than one model. I still include all of them, because of guard at line 103, which prevent from having more than one.

In the future, when we will support than one, we will already have the proper logic to handle multiples, as we will simply remove the guard at line 103.